data_store/messaging/tf_messaging.py

```diff
@@ -447,6 +454,9 @@ def _get_cell_indexes_for_outcomes(self, failed_row: pd.Series) -> str:
         financial_year = self._get_uk_financial_year_start(start_date)
         index = failed_row.name
 
+        if not isinstance(index, int):
+            raise TypeError(f"Cell index not int for failed row {failed_row}")
```

Comment:
Looking at the tests we seem to assume `index` is always an `int`. But it seems to be returning a `Hashable | None`.

Reply:
Yeah I think pandas treats a few things as just 'Hashables' - but I think in reality maybe one of our code assumptions is that it is _actually_ always an `int` as we use the library. Possibly something we can fix with tighter type hints somewhere ...